services/migrations/github_exported_data.go

```diff
@@ -0,0 +1,1335 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"code.gitea.io/gitea/modules/json"
+	"code.gitea.io/gitea/modules/log"
+	base "code.gitea.io/gitea/modules/migration"
+
+	"github.com/hashicorp/go-version"
+)
+
+/*
+	{"version":"1.0.1","github_sha":"8de0984858fd99a8dcd2d756cf0f128b9161e3b5"}
+*/
+type githubSchema struct {
+	Version   string
+	GithubSha string `json:"github_sha"`
+}
+
+/*
+{
+    "type": "user",
+    "url": "https://github.com/sunvim",
+    "avatar_url": "https://avatars.githubusercontent.com/u/859692?v=4",
+    "login": "sunvim",
+    "name": "mobus",
+    "bio": "code happy ",
+    "company": "Ankr",
+    "website": null,
+    "location": "Shanghai",
+    "emails": [
+      {
+        "address": "sv0220@163.com",
+        "primary": true,
+        "verified": true
+      }
+    ],
+    "billing_plan": null,
+    "created_at": "2011-06-19T11:25:35Z"
+  },
+*/
+type githubUser struct {
+	URL       string
+	AvatarURL string `json:"avatar_url"`
+	Login     string
+	Name      string
+	Bio       string
+	Company   string
+	Website   string
+	Location  string
+	Emails    []struct {
+		Address  string
+		Primary  bool
+		Verified bool
+	}
+	CreatedAt time.Time `json:"created_at"`
+}
+
+func getID(s string) int64 {
+	u, _ := url.Parse(s)
+	fields := strings.Split(u.Path, "/")
+	i, _ := strconv.ParseInt(fields[len(fields)-1], 10, 64)
+	return i
+}
+
+func (g *githubUser) ID() int64 {
+	return getID(g.AvatarURL)
+}
+
+func (g *githubUser) Email() string {
+	if len(g.Emails) < 1 {
+		return ""
+	}
+
+	for _, e := range g.Emails {
+		if e.Primary {
+			return e.Address
+		}
+	}
+	return ""
+}
+
+/*{
+    "type": "attachment",
+    "url": "https://user-images.githubusercontent.com/1595118/2923824-63a167ce-d721-11e3-91b6-74b83dc345bb.png",
+		"issue": "https://github.com/go-xorm/xorm/issues/205",
+    "issue_comment": "https://github.com/go-xorm/xorm/issues/115#issuecomment-42628488",
+    "user": "https://github.com/mintzhao",
+    "asset_name": "QQ20140509-1.2x.png",
+    "asset_content_type": "image/png",
+    "asset_url": "tarball://root/attachments/63a167ce-d721-11e3-91b6-74b83dc345bb/QQ20140509-1.2x.png",
+    "created_at": "2014-05-09T02:38:54Z"
+  },
+*/
+type githubAttachment struct {
+	Issue            string
```

Comment:
What is this field for? No sample data in above structure. There is a `type`

Reply:
The third field of the example is issue. We can know if the issue index and know if it's a pull request.